src/jit/codegenlegacy.cpp

```diff
@@ -19047,16 +19047,30 @@ regMaskTP CodeGen::genCodeForCall(GenTreeCall* call, bool valUsed)
                     {
                         noway_assert(callType == CT_USER_FUNC);
 
-                        void* pAddr;
-                        addr = compiler->info.compCompHnd->getAddressOfPInvokeFixup(methHnd, (void**)&pAddr);
-                        if (addr != NULL)
+                        CORINFO_CONST_LOOKUP lookup;
+                        compiler->info.compCompHnd->getAddressOfPInvokeTarget(methHnd, &lookup);
+
+                        void* addr = lookup.addr;
+
+                        assert(addr != NULL);
+
+#if defined(_TARGET_ARM_)
+                        // Legacy backend does not handle the `IAT_VALUE` case that does not
+                        // fit. It is not reachable currently from any front end so just check
+                        // for it via noway_assert.
+                        noway_assert(lookup.accessType != IAT_VALUE || compiler->codeGen->validImmForBL((ssize_t)addr));
+#endif
```

Comment:
Change `compiler->codeGen->validImmForBL((ssize_t)addr)` to `arm_Valid_Imm_For_BL((ssize_t)addr)`